jetty-server/src/main/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandler.java

```diff
@@ -0,0 +1,376 @@
+                    {
+                        String ct = response.getContentType();
+                        if (ct == null)
+                            _aggregating = Boolean.TRUE;
+                        else
+                        {
+                            ct = MimeTypes.getContentTypeWithoutCharset(ct);
+                            _aggregating = isMimeTypeBufferable(StringUtil.asciiToLowerCase(ct));
+                        }
+                    }
+                }
+            }
+
+            // If we are not aggregating, then handle normally.
+            if (!_aggregating)
+            {
+                getNextInterceptor().write(content, last, callback);
+                return;
+            }
+
+            if (LOG.isDebugEnabled())
+                LOG.debug("{} aggregating", this);
+
+            try
+            {
+                aggregate(content);
+            }
+            catch (Throwable t)
+            {
+                resetBuffer();
+                callback.failed(t);
+                return;
+            }
+
+            if (last)
+                commit(callback);
+            else
+                callback.succeeded();
+        }
+
+        protected void aggregate(ByteBuffer content) throws IOException
+        {
+            if (_bufferedOutputStream == null)
+            {
+                // Create a new OutputStream to a file.
+                _file = Files.createTempFile(tempDir, "BufferedResponse", "").toFile();
+                _bufferedOutputStream = new BufferedOutputStream(Files.newOutputStream(_file.toPath(), StandardOpenOption.WRITE));
+            }
+
+            BufferUtil.writeTo(content, _bufferedOutputStream);
+        }
+
+        @Override
+        public Interceptor getNextInterceptor()
+        {
+            return _next;
+        }
+
+        protected void commit(Callback callback)
+        {
+            try
+            {
+                _bufferedOutputStream.flush();
+                _bufferedOutputStream.close();
+                _bufferedOutputStream = null;
+            }
+            catch (IOException e)
+            {
+                resetBuffer();
+                callback.failed(e);
+                return;
+            }
+
+            // Create an iterating callback to do the writing
+            IteratingCallback icb = new IteratingCallback()
+            {
+                private final long fileLength = _file.length();
+                private int _pos = 0;
+                private ByteBuffer _buffer;
+                private boolean _last = false;
+
+                @Override
+                protected Action process() throws Exception
+                {
+                    if (_last)
+                        return Action.SUCCEEDED;
+
+                    long len = Math.min(_outputBufferSize, fileLength - _pos);
+                    _last = (_pos + len == fileLength);
+                    _buffer = BufferUtil.toMappedBuffer(_file, _pos, len);
```

Comment:
Did your research find any recent info on the memory impact of making temporary FileMappedBuffers?    
Also would be good to actually test that to make sure there are no memory impacts of creating then discarding many FileMappedBuffers.  @sbordet do you know the latest on this?

Reply:
@gregw There's no good way to just unmap the MappedByteBuffer until it is garbage collected. I talked to @sbordet about this and he suggested the Cleaner API be used to clean up the buffers when the Iterating Callback is completed each time. But that is only available since Java 9 so we won't be able to use it on 9.4.

Reply:
We may have to give some users a version of this that works in 9.4, but uses the java9 Cleaner APIs.   When you do a PR to 10 for this, add those APIs and then we can cherry pick those changes to specific 9.4 branches if needed. @joakime  thoughts?

Reply:
If FileMappedBuffers have the behavior that @sbordet mentioned, then Jetty 9.4.x wouldn't be able to use them as-is.
Especially for the Java 8 users out there.

Reply:
If the requested resource winds up in DefaultServlet / ResourceService and those are serving a static resource via MappedMemory as well, wouldn't this cause extra problems?

Reply:
@lachlan-roberts Can you write a test that does lots and lots of randomly generated responses of various sizes and then monitor what happens with memory for them running java8 and other JVMs if you leave them running for several hours.  Also test how it goes on JVM >= 9 using the API referred to by @sbordet.

If it is the case that on java8 we always run out of memory, then we can't use file mapped buffers like this and might have to revert (in 9 but not 10) to cycling through a single fixed sized indirect buffer.   
Another option is to make this a java >=9 feature, as we already have some of those in jetty-9 (ALPN was one for example).  @joakime is the sponsor using java >= 9?